### PR TITLE
micro-updates to documentation development, on python-poetry

### PR DESCRIPTION
**Pull request by lersek**
- `docs/Makefile`: work around python-poetry issue https://github.com/python-poetry/poetry/issues/8761
- `docs/README.md`: fix minimum poetry version

No backporting needed (docs development).

### CODE REVIEW

**Comment by lersek** [comment]
@annastuchlik @denesb please check this; thanks!

**Comment by tchaikov** [comment]
@dgarcia360 hi David, could you take a look as well?

File: docs/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Global variables
 # You can set these variables from the command line.
-POETRY        := poetry
+POETRY        := PYTHON_KEYRING_BACKEND=keyring.backends.null.Keyring poetry
```
Comment (tchaikov) [inline docs/Makefile:3]: @lersek i'd suggest just put
```make
export PYTHON_KEYRING_BACKEND = keyring.backends.null.Keyring
```
before `POETRY := poetry` . see also https://www.gnu.org/software/make/manual/html_node/Variables_002fRecursion.html

more readable this way.

**Review by tchaikov — APPROVED** [review]
lgtm

File: docs/Makefile (hunk 1 above)
Comment (lersek) [reply docs/Makefile:3]: @tchaikov Hi Kefu, it seems like I missed your comment here; not sure how.

I think I prefer the env var being set only for poetry (and its children). Overriding the user's keyring backend is generally unjustified; this is only a workaronud for the poetry bug, so I'd like to keep it as narrowly scoped as possible. IIUC, the `export` would apply the `PYTHON_KEYRING_BACKEND` setting to anything that make runs; that's not my intent.

**Pull request merged**